libyul/ControlFlowSideEffectsCollector.cpp

```diff
@@ -0,0 +1,272 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libyul/ControlFlowSideEffectsCollector.h>
+
+#include <libyul/optimiser/FunctionDefinitionCollector.h>
+
+#include <libyul/AST.h>
+#include <libyul/Dialect.h>
+
+#include <libsolutil/Common.h>
+#include <libsolutil/Algorithms.h>
+
+#include <range/v3/view/reverse.hpp>
+
+using namespace std;
+using namespace solidity::yul;
+
+void ControlFlowBuilder::operator()(FunctionCall const& _functionCall)
+{
+	walkVector(_functionCall.arguments | ranges::views::reverse);
+	newConnectedNode();
+	m_currentNode->functionCall = {_functionCall.functionName.name};
```

Comment:
```suggestion
	m_currentNode->functionCall = _functionCall.functionName.name;
```
The braces are a bit misleading for the optional type.